File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/TaxRateHistory.js
/**
 * Historical tax rate information for a tax.
 */

function TaxRateHistory() {

    this.EffectiveDate = new Date(0);
    this.LevelExemptible = false;
    this.BracketInfo = [];

    /**
     * Sets the effective date for the tax rate.
     * @param effectiveDate Date
     * @return Reference to this object for method chaining. 
     */
    this.effectiveDate = function (effectiveDate) {
        this.EffectiveDate = effectiveDate;
        return this;
    }

    /**
     * Effective date for rate.
     * @return Date
     */
    this.getEffectiveDate = function () {
        return this.EffectiveDate;
    }

    /* Reference to this object for method chaining.*
     * Sets the effective date for the tax rate.
     * @param effectiveDate Date 
     */
    this.setEffectiveDate = function (effectiveDate) {
        this.EffectiveDate = effectiveDate;
    }

    /**
     * Sets a value indicating if the tax can be exempted using level exemptions.
     * @param levelExemptible Boolean
     * @return Reference to this object for method chaining. 
     */
    this.levelExemptible = function (levelExemptible) {
        this.LevelExemptible = levelExemptible;
        return this;
    }

    /**
     * Indicates if the tax can be exempted using level exemptions.
     * @return Boolean
     */
    this.getLevelExemptible = function () {
        return this.LevelExemptible;
    }

    /* Reference to this object for method chaining.*
     * Sets a value indicating if the tax can be exempted using level exemptions.
     * @param levelExemptible Boolean 
     */
    this.setLevelExemptible = function (levelExemptible) {
        this.LevelExemptible = levelExemptible;
    }

    /**
     * Sets the list of tax rate or bracket information.
     * @param bracketInfo Array of TaxBracketInfo
     * @return Reference to this object for method chaining. 
     */
    this.bracketInfo = function (bracketInfo) {
        this.BracketInfo = bracketInfo;
        return this;
    }

    /**
     * Adds a bracket to this tax rate history record.
     * @param bracketInfoItem TaxBracketInfo object.
     * @return Reference to this object for method chaining. 
     */
    this.addBracketInfoItem = function (bracketInfoItem) {
        this.BracketInfo.push(bracketInfoItem);
        return this;
    }

    /**
     * Tax rate/bracket information.
     * @return Array of TaxBracketInfo
     */
    this.getBracketInfo = function () {
        return this.BracketInfo;
    }

    /* Reference to this object for method chaining.*
     * Sets the list of tax rate or bracket information.
     * @param bracketInfo Array of TaxBracketInfo 
     */
    this.setBracketInfo = function (bracketInfo) {
        this.BracketInfo = bracketInfo;
    }
}

File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/TaxRateHistory.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

/**
 * Historical tax rate information for a tax.
 */
public class TaxRateHistory {

    @SerializedName("EffectiveDate")
    private LocalDateTime effectiveDate = null;

    @SerializedName("LevelExemptible")
    private Boolean levelExemptible = null;

    @SerializedName("BracketInfo")
    private List<TaxBracketInfo> bracketInfo = new ArrayList<TaxBracketInfo>();

    /**
     * Sets the effective date for the tax rate.
     * @param effectiveDate
     * @return 
     */
    public TaxRateHistory effectiveDate(LocalDateTime effectiveDate) {
        this.effectiveDate = effectiveDate;
        return this;
    }

    /**
     * Effective date for rate.
     * @return effectiveDate
     */
    public LocalDateTime getEffectiveDate() {
        return effectiveDate;
    }

    /**
     * Sets the effective date for the tax rate.
     * @param effectiveDate 
     */
    public void setEffectiveDate(LocalDateTime effectiveDate) {
        this.effectiveDate = effectiveDate;
    }

    /**
     * Sets a value indicating if the tax can be exempted using level exemptions.
     * @param levelExemptible
     * @return 
     */
    public TaxRateHistory levelExemptible(Boolean levelExemptible) {
        this.levelExemptible = levelExemptible;
        return this;
    }

    /**
     * Indicates if the tax can be exempted using level exemptions.
     * @return levelExemptible
     */
    public Boolean getLevelExemptible() {
        return levelExemptible;
    }

    /**
     * Sets a value indicating if the tax can be exempted using level exemptions.
     * @param levelExemptible 
     */
    public void setLevelExemptible(Boolean levelExemptible) {
        this.levelExemptible = levelExemptible;
    }

    /**
     * Sets the list of tax rate or bracket information.
     * @param bracketInfo
     * @return 
     */
    public TaxRateHistory bracketInfo(List<TaxBracketInfo> bracketInfo) {
        this.bracketInfo = bracketInfo;
        return this;
    }

    /**
     * Adds a bracket to this tax rate history record.
     * @param bracketInfoItem
     * @return 
     */
    public TaxRateHistory addBracketInfoItem(TaxBracketInfo bracketInfoItem) {
        this.bracketInfo.add(bracketInfoItem);
        return this;
    }

    /**
     * Tax rate/bracket information.
     * @return bracketInfo
     */
    public List<TaxBracketInfo> getBracketInfo() {
        return bracketInfo;
    }

    /**
     * Sets the list of tax rate or bracket information.
     * @param bracketInfo 
     */
    public void setBracketInfo(List<TaxBracketInfo> bracketInfo) {
        this.bracketInfo = bracketInfo;
    }
}